### PR TITLE
Remove snap CI Provider, add GoCD provider

### PR DESCRIPTION
**Pull request by sog01**
Fix issue #2680

Remove snap in CI_PROVIDERS const and replace it with goCD, also add GoCD environment variable(based on https://docs.gocd.org/current/faq/environment_variables.html#standard-gocd-environment-variables)

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
There are some failing tests in `packages/server/test/unit/ci_provider_spec.coffee` that are checking for 'snap'. Need these removed and added test for GoCD. 

Some snapshots need updating, where we list the CI Providers in the CLI also. `packages/server/test/integration/cypress_spec.coffee`

May be some other test failures.

**Comment by sog01** [comment]
Ok, I will fix it

**Comment by jennifer-shehane** [comment]
@sog01 Yes, follow the instructions for contributing to the `server` package here: https://github.com/cypress-io/cypress/blob/develop/packages/server/README.md

File: packages/server/lib/util/ci_provider.js [hunk 1]
```diff
@@ -222,7 +222,27 @@ const _providerCiParams = () => {
...
+      'GO_SERVER_URL',
+      'GO_ENVIRONMENT_NAME',
+      'GO_PIPELINE_NAME',
+      'GO_PIPELINE_COUNTER',
+      'GO_PIPELINE_LABEL',
+      'GO_STAGE_NAME',
+      'GO_STAGE_COUNTER',
+      'GO_JOB_NAME',
+      'GO_TRIGGER_USER',
+      'GO_DEPENDENCY_LABEL_${pipeline_name}',
```
Comment (bahmutov) [inline packages/server/lib/util/ci_provider.js]: hmm, are these dynamically set?

Comment (sog01) [reply packages/server/lib/util/ci_provider.js]: Sorry for late reply, I have busy work that need to handle it first. So, we need to change it into **GO_DEPENDENCY_LABEL** ?. Currently, there are no example, to write dynamic variable.

**Pull request closed without merging**